src/linkgraph/linkgraph_gui.h

```diff
@@ -109,7 +109,7 @@ struct LinkGraphLegendWindow : Window {
 	void OnInvalidateData(int data = 0, bool gui_scope = true) override;
 
 private:
-	LinkGraphOverlay *overlay;
+	std::shared_ptr<LinkGraphOverlay> overlay;
```

Comment:
I wasn't sure about it, because there are other parts of the system that appear to take references to viewport overlays; so I took a better-safe-than-sorry approach.

It's likely this entire system will be rewritten to use std::unique_ptr (or, possibly, the pool allocator?) to alleviate the needless overhead of weak pointers.